controllers/datadogagent/feature/logcollection/feature.go

```diff
@@ -0,0 +1,171 @@
+	logCollection := dda.Spec.Features.LogCollection
+
+	if logCollection != nil && apiutils.BoolValue(logCollection.Enabled) {
+		f.enable = true
+		if logCollection.ContainerCollectAll != nil && apiutils.BoolValue(logCollection.ContainerCollectAll) {
+			f.containerCollectAll = true
+		}
+		if apiutils.BoolValue(logCollection.ContainerCollectUsingFiles) {
+			f.containerCollectUsingFiles = true
+		}
+		f.containerLogsPath = *logCollection.ContainerLogsPath
+		f.podLogsPath = *logCollection.PodLogsPath
+		f.containerSymlinksPath = *logCollection.ContainerSymlinksPath
+		f.tempStoragePath = *logCollection.TempStoragePath
+		if logCollection.OpenFilesLimit != nil {
+			f.openFilesLimit = *logCollection.OpenFilesLimit
+		}
+
+		reqComp = feature.RequiredComponents{
+			Agent: feature.RequiredComponent{
+				IsRequired: &f.enable,
+				Containers: []apicommonv1.AgentContainerName{
+					apicommonv1.CoreAgentContainerName,
+				},
+			},
+		}
+	}
+	return reqComp
+}
+
+// ConfigureV1 use to configure the feature from a v1alpha1.DatadogAgent instance.
+func (f *logCollectionFeature) ConfigureV1(dda *v1alpha1.DatadogAgent) (reqComp feature.RequiredComponents) {
+	logCollection := dda.Spec.Features.LogCollection
+
+	if logCollection != nil && apiutils.BoolValue(logCollection.Enabled) {
+		f.enable = true
+		if apiutils.BoolValue(logCollection.LogsConfigContainerCollectAll) {
+			f.containerCollectAll = true
+		}
+		if apiutils.BoolValue(logCollection.ContainerCollectUsingFiles) {
+			f.containerCollectUsingFiles = true
+		}
+		f.containerLogsPath = *logCollection.ContainerLogsPath
+		f.podLogsPath = *logCollection.PodLogsPath
+		f.containerSymlinksPath = *logCollection.ContainerSymlinksPath
+		f.tempStoragePath = *logCollection.TempStoragePath
+		if logCollection.OpenFilesLimit != nil {
+			f.openFilesLimit = *logCollection.OpenFilesLimit
+		}
+
+		reqComp = feature.RequiredComponents{
+			Agent: feature.RequiredComponent{
+				IsRequired: &f.enable,
+				Containers: []apicommonv1.AgentContainerName{
+					apicommonv1.CoreAgentContainerName,
+				},
+			},
+		}
+	}
+	return reqComp
+}
+
+// ManageDependencies allows a feature to manage its dependencies.
+// Feature's dependencies should be added in the store.
+func (f *logCollectionFeature) ManageDependencies(managers feature.ResourceManagers) error {
+	return nil
+}
+
+// ManageClusterAgent allows a feature to configure the ClusterAgent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *logCollectionFeature) ManageClusterAgent(managers feature.PodTemplateManagers) error {
+	return nil
+}
+
+// ManageNodeAgent allows a feature to configure the Node Agent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *logCollectionFeature) ManageNodeAgent(managers feature.PodTemplateManagers) error {
+	if f.enable {
```

Comment:
hmm good thought; i imagine we'll want to put this one level up, but it's not present yet (in the cluster agent reconciler)

Reply:
just to close the loop on this, we discussed separately but the features are filtered using the RequiredComponents struct a couple levels up. so this isn't needed and it may be possible to get rid of f.enable altogether